SU2_CFD/src/solvers/CSpeciesSolver.cpp

```diff
@@ -0,0 +1,724 @@
+      cout << "Compute linelet structure. " << nLineLets << " elements in each line (average)." << endl;
+  }
+
+  LinSysSol.Initialize(nPoint, nPointDomain, nVar, 0.0);
+  LinSysRes.Initialize(nPoint, nPointDomain, nVar, 0.0);
+  System.SetxIsZero(true);
+
+  if (ReducerStrategy) EdgeFluxes.Initialize(geometry->GetnEdge(), geometry->GetnEdge(), nVar, nullptr);
+
+  /*--- Initialize the BGS residuals in multizone problems. ---*/
+  if (multizone) {
+    Residual_BGS.resize(nVar, 0.0);
+    Residual_Max_BGS.resize(nVar, 0.0);
+    Point_Max_BGS.resize(nVar, 0);
+    Point_Max_Coord_BGS.resize(nVar, nDim) = su2double(0.0);
+  }
+
+  //} //iMESH_0
+
+  /*--- Initialize lower and upper limits---*/
+
+  if (config->GetSpecies_Clipping()) {
+    for (auto iVar = 0u; iVar < nVar; iVar++) {
+      lowerlimit[iVar] = config->GetSpecies_Clipping_Min(iVar);
+      upperlimit[iVar] = config->GetSpecies_Clipping_Max(iVar);
+    }
+  } else {
+    for (auto iVar = 0u; iVar < nVar; iVar++) {
+      lowerlimit[iVar] = -1.0e15;
+      upperlimit[iVar] = 1.0e15;
+    }
+  }
+  /*--- Far-field flow state quantities and initialization. ---*/
+  // su2double Density_Inf, Viscosity_Inf;
+  // Density_Inf   = config->GetDensity_FreeStreamND();
+  // Viscosity_Inf = config->GetViscosity_FreeStreamND();
+
+  /*--- Set up fluid model for the diffusivity ---*/
+
+  su2double Diffusivity_Ref = 1.0;
+  su2double DiffusivityND = config->GetDiffusivity_Constant() / Diffusivity_Ref;
+  config->SetDiffusivity_ConstantND(DiffusivityND);
+
+  /*--- Scalar variable state at the far-field. ---*/
+
+  for (auto iVar = 0u; iVar < nVar; iVar++) {
+    Species_Inf[iVar] = config->GetSpecies_Init(iVar);
+  }
+
+  /*--- Initialize the solution to the far-field state everywhere. ---*/
+
+  nodes = new CSpeciesVariable(Species_Inf, nPoint, nDim, nVar, config);
+  SetBaseClassPointerToNodes();
+
+  /// NOTE TK:: This should use a MassDiff model!
+  /*--- initialize the mass diffusivity ---*/
+  SU2_OMP_FOR_STAT(omp_chunk_size)
+  for (unsigned long iPoint = 0; iPoint < nPoint; iPoint++) {
+    for (auto iVar = 0u; iVar < nVar; iVar++) {
+      const auto massDiffusivity = config->GetDiffusivity_Constant();  // TK this should be ND
+      nodes->SetDiffusivity(iPoint, massDiffusivity, iVar);
+    }
+  }
+  END_SU2_OMP_FOR
+
+  /*--- MPI solution ---*/
+
+  InitiateComms(geometry, config, SOLUTION);
+  CompleteComms(geometry, config, SOLUTION);
+
+  /// NOTE TK:: Hinder sliding mesh in cfg postprocessing
+
+  /*-- Allocation of inlets has to happen in derived classes
+   (not CScalarLegacySolver), due to arbitrary number of scalar variables.
+   First, we also set the column index for any inlet profiles. ---*/
+
+  Inlet_Position = nDim + 2 + nDim; // Coords(nDim), T(1), velocity_mag(1), Normal(nDim), Turb(1 or 2), species
+  if (config->GetKind_Turb_Model() == TURB_MODEL::SA)
+    Inlet_Position += 1;
+  else if (config->GetKind_Turb_Model() == TURB_MODEL::SST)
+    Inlet_Position += 2;
```

Comment:
This will not work with many turb models right :)
In addressing it please add some function in CConfig to get the number of turb variables.

Reply:
True, ... we indeed have no `nTurb(Var) ` style variable in the config. I'll add one